test/endpoints/conftest.py

```diff
@@ -0,0 +1,116 @@
+import asyncio
+
+from fastapi.testclient import TestClient
+import pytest
+
+from operationsgateway_api.src.main import app
+from operationsgateway_api.src.mongo.interface import MongoDBInterface
+
+
+@pytest.fixture()
+def test_app():
+    return TestClient(app)
+
+
+@pytest.fixture()
+def login_and_get_token(test_app: TestClient):
+    json = '{"username": "backend", "password": "back"}'
+    response = test_app.post("/login", content=json)
+    # strip the first and last characters off the response
+    # (the double quotes that surround it)
+    token = response.text[1:-1]
+    return token
+
+
+@pytest.fixture()
+def loop():
+    # allows the testing of asynchronus functions using an event loop
+
+    loop = asyncio.new_event_loop()
+    yield loop
+    loop.close()
```

Comment:
The reason none of the fixtures use asyncio and instead use the event loop is because I tried them with async but nothing worked I looked up why and I don't think the pytests can use async fixtures.
But i'll give it another try anyway

Reply:
I found the pytest_asyncio library which nicely allows the pytest to use async functions and have implamented it (however it does mean a new dependency is required if thats an issue)

Reply:
`pytest-asyncio` is already a dependency of this project so that's not a problem. Search for `@pytest.mark.asyncio` to see some examples of where we already use this. In fact, the library is listed as a regular dependency as well as a development dependency - as it's a testing library, it should only be a dev dependency. I'll create an issue on Jira for this

Reply:
ok great. it works with the `@pytest_asyncio.fixture(scope="function")`